### DISCUSSION TITLE
How to get all available far field radiation spheres setups

### DISCUSSION THREAD

**Discussion by AleksandarToshev-eag** [ansys/pyaedt#4943] [Q&A] [answered]
Hi Everybody, I am asking for a help (advise) on how to get all available far field radiation setups in a HFSS design. I am building a GUI, where I couple to an unknown project so I need to read the names of all available far field setups and to select a particular one for further processing. Any help would be welcome (I was checking the Pyaedt documentation vaguely and could not find a get method on far field setup. For example I have like Hfss.get_setups(), but I cannot find get method for far field radiation spheres)

Thanks in advance for the help!

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @AleksandarToshev-eag ,

You can get the radiation spheres using field_setups property:

hfss = pyaedt.Hfss(design="Horn_ffd1")
ff_setups = hfss.field_setups

Where ff_setups is a list of objects:
ff_setups[0].name

If you just need the names:
hfss.field_setup_names

Please do not forget to save the design before getting this information, if you have created the ff setups manually (without pyaedt) the object could be not updated. If this solves your question, please mark it as an answer and close the discussion.

**Reply by AleksandarToshev-eag** [reply]
Thank you Samuel!
